url29ecd5d728_geodb_db3dcore_tgz-pJ8-de_uos_igf_db3d_dbms_model3d_SegmentNetBuilderJ8/src/de/uos/igf/db3d/dbms/model3d/TetrahedronNet3DComp.java

```diff
@@ -0,0 +1,2632 @@
+				for (int i = 0; i < 4; i++) {
+					if (!tet.hasNeighbour(i))
+						hull.add(tet.getTriangle(i));
+				}
+		}
+
+		it = hull.iterator();
+		while (it.hasNext()) {
+			Triangle3D tri = (Triangle3D) it.next();
+			if (tetra.intersects(tri, this.getScalarOperator()))
+				// Here an IllegalStateException can be thrown signaling
+				// problems with the dimensions of the wireframe.
+				return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Return the type of this as a <code>ComplexGeoObj</code>.
+	 * 
+	 * @return byte - type.
+	 * @see db3d.dbms.model3d.ComplexGeoObj#getType()
+	 */
+	public byte getType() {
+		return ComplexGeoObj.COMP_TETRAHEDRON_NET_3D;
+	}
+
+	/**
+	 * Returns the enclosing net
+	 * 
+	 * @return TetrahedronNet3D - enclosing net.
+	 */
+	public TetrahedronNet3D getNet() {
+		return net;
+	}
+
+	/**
+	 * Performs a deep copy of this component with all its recursive members.<br>
+	 * Only the reference to the enclosing net is not copied and must be set
+	 * afterwards.
+	 * 
+	 * @return TetrahedronNet3DComp - deep copy.
+	 */
+	public TetrahedronNet3DComp serializationCopy() {
+		try {
+			ByteArrayOutputStream out = new ByteArrayOutputStream();
+			ObjectOutputStream os = new ObjectOutputStream(out);
+			os.writeObject(this);
+			os.flush();
+			ObjectInputStream is = new ObjectInputStream(
+					new ByteArrayInputStream(out.toByteArray()));
+			Object ret = is.readObject();
+			is.close();
+			os.reset();
+			os.close();
+			return (TetrahedronNet3DComp) ret;
+		} catch (IOException e) {
+			e.printStackTrace();
+			return null;
+		} catch (ClassNotFoundException e) {
+			e.printStackTrace();
+			return null;
+		}
+	}
+
+	/**
+	 * Sets the reference to enclosing TetrahedronNet3D.
+	 * 
+	 * @param net3D
+	 *            enclosing TetrahedronNet3D
+	 */
+	public void setNet(TetrahedronNet3D net3D) {
+		this.net = net3D;
+	}
+
+	/*
+	 * Load the sam at construction time and counts the vertices.
+	 * 
+	 * @param elements TetrahedronElt3D whose elements should be inserted into
+	 * SAM
+	 * 
+	 * @throws IllegalArgumentException if an attempt is made to construct a
+	 * MBB3D whose maximum point is not greater than its minimum point.
+	 */
+	private void loadSAM(TetrahedronElt3D[] elements) {
+		for (int i = 0; i < elements.length; i++)
+			this.sam.insert(elements[i]);
+	}
+
+	/**
+	 * Builds the neighbour topology of the net for the given Tetrahedron
+	 * elements.
+	 * 
+	 * @param elts
+	 *            TetrahedronElt3D[]
+	 * @throws DB3DException
+	 *             - during registering neighbours, a DB3DException is thrown if
+	 *             the neighbour index is not 0, 1, 2 or 3.
+	 * @throws IllegalArgumentException
+	 *             if the index of the point of the tetrahedron is not in the
+	 *             interval [0;3]. The exception originates in the method
+	 *             getPoint(int) of the class Tetrahedron3D.
+	 */
+	public void buildNetTopology(TetrahedronElt3D[] elts) throws DB3DException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 29. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.